python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,258 @@
+
+from lsst.ts.cRIOpy.M1M3FATable import (
+    FATABLE,
+    FATABLE_ID,
+)
+
+
+class CheckActuators(BaseScript):
+    """Perform a M1M3 bump test on either a selection of individual
+    actuators or on all actuators..
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Running bump test on FA ID: {id}.": Check individual actuator.
+    - "M1M3 bump test completed.": Check complete.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(index=index, descr="Bump Test on M1M3 Actuators")
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+        # Average duration (seconds) of a bump test on a single actuator
+        self.time_one_bump = 25
+
+        # Getting list of actuator ids from mtcs
+        self.m1m3_actuator_ids = self.mtcs.get_m1m3_actuator_ids()
+        self.m1m3_secondary_actuator_ids = self.mtcs.get_m1m3_actuator_secondary_ids()
+
+        # Actuators that will be effectively tested
+        self.actuators_to_test = self.m1m3_actuator_ids.copy()
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state before
+        running bump test.
+        """
+
+        for comp in self.mtcs.components_attr:
+            if comp != "mtm1m3":
+                self.log.debug(f"Ignoring component {comp}.")
+                setattr(self.mtcs.check, comp, False)
+
+        # Check all enabled, liveliness and the m1m3 detailed state
+        await asyncio.gather(
+            self.mtcs.assert_all_enabled(),
+            self.mtcs.assert_liveliness(),
+            self.mtcs.assert_m1m3_detailed_state(
+                {
+                    DetailedState.PARKED,
+                    DetailedState.PARKEDENGINEERING,
+                    DetailedState.ACTIVE,
+                    DetailedState.ACTIVEENGINEERING,
+                }
+            ),
+        )
+
+    @classmethod
+    def get_schema(cls):
+        m1m3_actuator_ids_str = ",".join([str(fa[FATABLE_ID]) for fa in FATABLE])
+
+        url = "https://github.com/lsst-ts/"
+        path = (
+            "ts_externalscripts/blob/main/python/lsst/ts/standardscripts/"
+            "maintel/m1m3/check_actuators.py"
+        )
+        schema_yaml = f"""
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: {url}{path}
+        title: CheckAcutators v1
+        description: Configuration for Maintel bump test SAL Script.
+        type: object
+        properties:
+            actuators:
+                description: Actuators to run the bump test.
+                oneOf:
+                  - type: array
+                    items:
+                      type: number
+                      enum: [{m1m3_actuator_ids_str}]
```

Comment:
How you distinguish between primary and secondary FAs? It might be a valid case. For the moment, add to description something like "Both cylinders will be tested for DAA."